plutus-tx/src/Language/PlutusTx/Evaluation.hs

```diff
@@ -0,0 +1,34 @@
+module Language.PlutusTx.Evaluation (evaluateCek, evaluateCekLog) where
+
+import           Language.PlutusCore.Interpreter.CekMachine hiding (evaluateCek)
+import           Language.PlutusCore
+import           Language.PlutusCore.Constant.Dynamic
+import           Language.PlutusCore.Constant
+
+import           Control.Exception
+
+import           System.IO.Unsafe
+
+stringBuiltins :: DynamicBuiltinNameMeanings
+stringBuiltins =
+    insertDynamicBuiltinNameDefinition dynamicCharToStringDefinition $ insertDynamicBuiltinNameDefinition dynamicAppendDefinition mempty
+
+-- TODO: pretty sure we shouldn't need the unsafePerformIOs here, we should expose a pure interface even if it has IO hacks under the hood
+
+-- | Evaluate a program in the CEK machine with the usual string dynamic builtins.
+evaluateCek
+    :: Program TyName Name ()
+    -> EvaluationResult
+evaluateCek p = unsafePerformIO $ evaluate $ runCek stringBuiltins p
+
+-- | Evaluate a program in the CEK machine with the usual string dynamic builtins and tracing, additionally
+-- returning the trace output.
+evaluateCekLog
```

Comment:
I think it's better then to use `trace` consistently everywhere rather than `log`.

Reply:
:+1: